### ISSUE TITLE
Add option to skip transactions

### ISSUE THREAD

**Issue by DiscountDarcy** [coldbox-modules/cfmigrations#23]
We recently had a very unusual edge case in which we had a migration that ran an `ALTER DATABASE` command that could not be processed inside a transaction.

We made a slight change to `MigrationService.cfc` to enable this:

```
               preProcessHook( migrationStruct );

		if ( structKeyExists( migration, 'transaction' ) && !migration.transaction ) {
			invoke(
				migration,
				direction,
				[ schema, query ]
			);
			logMigration( direction, migrationStruct.componentName );
		}
		else {
			transaction action="begin" {
				try {
					invoke(
						migration,
						direction,
						[ schema, query ]
					);
					logMigration( direction, migrationStruct.componentName );
				} catch ( any e ) {
					transaction action="rollback";
					rethrow;
				}
			}
		}

		postProcessHook( migrationStruct );
```

And then on the migration itself we just set `	this.transaction = false;`.

Would you take this as a PR?

**Comment by elpete** [comment]
Yeah. That would be fine. How would you feel about specifying it as an
attribute on the function (`up` or `down`) or even the component?

**Comment by DiscountDarcy** [comment]
On the function doesn't really help us since we tend to run all available migrations at once. We just put it on the component, as referenced by the example above, so the migration looks like this:

```
component accessors=true {

	this.transaction = false;

	function up( schema, query ) {
     ...

```

That seemed like the path of least resistance since anything else specific to the migration file would have to be a function or else extending a parent migration CFC, which didn't seem necessary just for this.

**Comment by elpete** [comment]
That looks fine. Call it `useTransactions` and we should be good.

**Comment by elpete** [comment]
Resolved in v3.0.0
